lib/class-wp-json-comments-controller.php

```diff
@@ -0,0 +1,470 @@
+	 * @return WP_Error|WP_JSON_Response|mixed
+	 */
+	public function get_item( $request ) {
+		$id = (int) $request['id'];
+
+		$comment = get_comment( $id );
+		if ( empty( $comment ) ) {
+			return new WP_Error( 'json_comment_invalid_id', __( 'Invalid comment ID.' ), array( 'status' => 404 ) );
+		}
+
+		if ( ! $this->check_read_permission( $comment ) ) {
+			return new WP_Error( 'json_user_cannot_read', __( 'Sorry, you cannot read this comment.' ), array( 'status' => 401 ) );
+		}
+
+		$post = get_post( $comment->comment_post_ID );
+		if ( empty( $post ) ) {
+			return new WP_Error( 'json_post_invalid_id', __( 'Invalid post ID.' ), array( 'status' => 404 ) );
+		}
+
+		if ( ! $this->check_read_post_permission( $post ) ) {
+			return new WP_Error( 'json_user_cannot_read', __( 'Sorry, you cannot read this post.' ), array( 'status' => 401 ) );
+		}
+
+		$data = $this->prepare_item_for_response( $comment, $request );
+		$response = json_ensure_response( $data );
+
+		return $response;
+	}
+
+	/**
+	 * Create a comment.
+	 *
+	 * @param  WP_JSON_Request $request Full details about the request.
+	 * @return WP_Error|WP_JSON_Response|mixed
+	 */
+	public function create_item( $request ) {
+		if ( ! empty( $request['id'] ) ) {
+			return new WP_Error( 'json_comment_exists', __( 'Cannot create existing comment.' ), array( 'status' => 400 ) );
+		}
+
+		$post = get_post( (int) $request['post_id'] );
+		if ( empty( $post ) ) {
+			return new WP_Error( 'json_post_invalid_id', __( 'Invalid post ID.' ), array( 'status' => 404 ) );
+		}
+
+		if ( ! $this->check_read_post_permission( $post ) ) {
+			return new WP_Error( 'json_user_cannot_read', __( 'Sorry, you cannot read this post.' ), array( 'status' => 401 ) );
+		}
+
+		if ( ! comments_open( $post->ID ) ) {
+			return new WP_Error( 'json_user_cannot_create', __( 'Sorry, the comments are closed for this post.' ), array( 'status' => 401 ) );
+		}
+
+		$prepared_comment = $this->prepare_item_for_database( $request );
+		$prepared_comment['comment_approved'] = wp_allow_comment( $prepared_comment );
+
+		$prepared_comment = apply_filters( 'json_pre_insert_comment', $prepared_comment, $request );
+
+		$comment_id = wp_insert_comment( $prepared_comment );
+		if ( ! $comment_id ) {
+			return new WP_Error( 'json_comment_failed_create', __( 'Creating comment failed.' ), array( 'status' => 500 ) );
+		}
+
+		$new_comment = get_comment( $comment_id );
+		$response = $this->prepare_item_for_response( $new_comment, array( 'context' => 'edit' ) );
+		$response = json_ensure_response( $response );
+
+		$response->set_status( 201 );
+		$response->header( 'Location', json_url( '/wp/comments/' . $comment_id ) );
+
+		return $response;
+	}
+
+	/**
+	 * Edit a comment
+	 *
+	 * @param  WP_JSON_Request $request Full details about the request.
+	 * @return WP_Error|WP_JSON_Response|mixed
+	 */
+	public function update_item( $request ) {
+		$id = (int) $request['id'];
+
+		$comment = get_comment( $id );
```

Comment:
@danielbachhuber This approach was the result of a discussion I had in our Slack channel with @joehoyle yesterday.  We cannot reuse the get_item method, if we want to allow unauthenticated users (user_id=0) the ability to create a comment.  
If an unauthenticated user creates a comment, the comment (depending on the comment moderation settings) will have a status of 'hold'.  The authenticated user cannot view the comment via the GET endpoint unless they have the `moderate_comments` permission.